### ISSUE TITLE
Guest Blog Post: CSS in JS babel plugins

### ISSUE THREAD

**Issue by hzoo** [babel/babel.github.io#1276]
I'd like to learn more about CSS in JS and why everyone is choosing to write a babel plugin to optimize code/make it faster

Maybe can collaborate on a post for the blog (if anyone is interested)?

cc:
https://github.com/zeit/styled-jsx: @rauchg?
https://github.com/styled-components/babel-plugin-styled-components: @philpl @mxstbr
https://github.com/tkh44/emotion @tkh44 @threepointone
others?

**Comment by philpl** [comment]
I'd love to talk about some insights here :smile: On the other hand, the styled-components preprocessing is only going to be stable and an official feature in v3. I actually have some slides on the topic here: https://sctalk.philpl.com/ (slide 20 & onward is about preprocessing)

Until then @tkh44 and @threepointone are definitely more of an authority to talk about this :smile:

I will actually be working on ISTF with @kof and then jump on a universal preprocessing lib that will also be a core lib for SC v3. So we could also write some things about ISTF, if you're interested, for now.

**Comment by kof** [comment]
By ISTF @philpl means this https://github.com/cssinjs/istf-spec

**Comment by hzoo** [comment]
I still think this would be cool to have! Would encourage others to try their own plugins, and maybe we could get some contributors to core 😄 (plz)

**Comment by mxstbr** [comment]
I can write something up about why we built ours initially and how it works. 😊

**Comment by giuseppeg** [comment]
@hzoo if time is not a constraint I can do that :)

**Comment by hzoo** [comment]
Yep, no time limits (believe me I understand that we don't always have time for these things)! Just a way to express the use cases for community plugins

**Comment by giuseppeg** [comment]
@hzoo maybe we can record an episode for the Babel Podcast? https://twitter.com/left_pad/status/1146069623940558849
